metecho/api/serializers.py

```diff
@@ -443,28 +452,40 @@ def get_branch_url(self, obj) -> Optional[str]:
         return None
 
     def get_branch_diff_url(self, obj) -> Optional[str]:
-        epic = obj.epic
-        epic_branch = epic.branch_name
-        project = epic.project
+        base_branch = obj.get_base()
+        project = obj.root_project
         repo_owner = project.repo_owner
         repo_name = project.repo_name
         branch = obj.branch_name
-        if repo_owner and repo_name and epic_branch and branch:
+        if repo_owner and repo_name and base_branch and branch:
             return (
                 f"https://github.com/{repo_owner}/{repo_name}/compare/"
-                f"{epic_branch}...{branch}"
+                f"{base_branch}...{branch}"
             )
         return None
 
     def get_pr_url(self, obj) -> Optional[str]:
-        project = obj.epic.project
+        project = obj.root_project
         repo_owner = project.repo_owner
         repo_name = project.repo_name
         pr_number = obj.pr_number
         if repo_owner and repo_name and pr_number:
             return f"https://github.com/{repo_owner}/{repo_name}/pull/{pr_number}"
         return None
 
+    def validate(self, data: dict) -> dict:
+        project = data.get("project", getattr(self.instance, "project", None))
+        epic = data.get("epic", getattr(self.instance, "epic", None))
+        if project and epic:
+            raise serializers.ValidationError(
+                _("Task can be attached to a Project or Epic, but not both")
+            )
+        if not (project or epic):
+            raise serializers.ValidationError(
+                _("Task must be attached to a Project or Epic")
+            )
+        return data
```

Comment:
This mimics what we already have in `models.py`, just to catch the error in validation instead of raising `IntegrityError` (which would be a 500)